.github/workflows/ci.yml

```diff
@@ -0,0 +1,47 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust: [stable, nightly]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.rust }}
+      - name: Run tests for all workspace crates
+        run: cargo test --workspace --all-features
```

Comment:
i need to use this rustfmt.toml:
```toml
edition = "2024"
error_on_line_overflow = true
error_on_unformatted = true

imports_granularity = "One"
use_small_heuristics = "Max"
use_field_init_shorthand = true

max_width = 80
tab_spaces = 2
```
but there are some nightly features (but it is just formatter). We need to check fmt in nightly channel probably